src/managed_scip.jl

```diff
@@ -0,0 +1,74 @@
+# holds pointers to SCIP data and takes care of memory mgmt
+mutable struct ManagedSCIP
+    scip::Ref{Ptr{SCIP_}}
+    vars::Vector{Ref{Ptr{SCIP_VAR}}}
+    conss::Vector{Ref{Ptr{SCIP_CONS}}}
+
+    function ManagedSCIP()
+        scip = Ref{Ptr{SCIP_}}()
+        @SC SCIPcreate(scip)
+        @assert scip[] != C_NULL
+        @SC SCIPincludeDefaultPlugins(scip[])
+        @SC SCIP.SCIPcreateProbBasic(scip[], "")
+
+        mscip = new(scip, [], [])
+        finalizer(free_scip, mscip)
+    end
+end
+
+"Release references and free memory"
+function free_scip(mscip::ManagedSCIP)
+    # avoid double-free
+    if mscip.scip[] != C_NULL
+        for cons in mscip.conss
+            @SC SCIPreleaseCons(mscip.scip[], cons)
+        end
+        for var in mscip.vars
+            @SC SCIPreleaseVar(mscip.scip[], var)
+        end
+        @SC SCIPfree(mscip.scip)
```

Comment:
Maybe add a note that this sets `mscip.scip` to `C_NULL` because it's not obvious.